android/src/main/java/com/iologinutils/AuthorizationManagerActivity.kt

```diff
@@ -0,0 +1,178 @@
+ *     </ul>
+ *   <li>S flow:
+ *     <ul>
+ *       <li>Step S1: The authorization activity completes with a success of failure, and sends
+ *           this result to {@link RedirectUriReceiverActivity}.
+ *       <li>Step S2: {@link RedirectUriReceiverActivity} extracts the forwarded data, and
+ *           invokes AuthorizationManagementActivity using an intent derived from
+ *           {@link #createResponseHandlingIntent}. This intent has flag CLEAR_TOP set, which
+ *           will result in both the authorization activity and
+ *           {@link RedirectUriReceiverActivity} being destroyed, if necessary, such that
+ *           AuthorizationManagementActivity is once again at the top of the back stack.
+ *       <li>Step S3: The Promise contained in the AuthorizationService is resolved with the
+ *  *        received data.
+ *     </ul>
+ * </ul>
+ */
+class AuthorizationManagerActivity : Activity() {
+
+  private var authorizationStarted = false
+  private var authIntent: Intent? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    intent.extras?.let {
+      extractState(it)
+    } ?: savedInstanceState?.let {
+      extractState(it)
+    } ?: throw IllegalStateException("No state to extract")
+  }
+
+  override fun onNewIntent(intent: Intent?) {
+    super.onNewIntent(intent)
+    setIntent(intent)
+  }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    outState.apply {
+      putBoolean(KEY_AUTHORIZATION_STARTED, authorizationStarted)
+      putParcelable(KEY_AUTH_INTENT, authIntent)
+    }
+  }
+
+  override fun onResume() {
+    super.onResume()
+
+    if (!authorizationStarted) {
+      try {
+        startActivity(authIntent)
+        authorizationStarted = true
+      } catch (e: ActivityNotFoundException) {
+        handleBrowserNotFound()
+        finish()
+      }
+      return
+    }
+
+    intent.data?.let { responseUri ->
+      handleAuthorizationComplete(responseUri)
+    } ?: run {
+      handleAuthorizationCanceled()
+    }
+
+    finish()
+  }
+
+  private fun handleAuthorizationComplete(responseUri: Uri) {
+    Log.d(
+      TAG,
+      "Authorization complete - invoking completion intent with $responseUri"
+    )
+    IoLoginUtilsModule.authorizationPromise?.resolve(responseUri.toString())
+  }
+
+  private fun handleAuthorizationCanceled() {
+    Log.d(TAG, "Authorization flow canceled by user")
+
+    IoLoginUtilsModule.authorizationPromise?.reject(
+      "NativeAuthSessionError",
+      generateErrorObject(IoLoginError.Type.NATIVE_AUTH_SESSION_CLOSED)
+    )
+  }
+
+  private fun extractState(state: Bundle) {
+    state.apply {
+      authIntent = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU) {
+        state.getParcelable(KEY_AUTH_INTENT, Intent::class.java)
+      } else {
+        @Suppress("DEPRECATION")
+        state.getParcelable(KEY_AUTH_INTENT)
+      }
```

Comment:
I handle the deprecation. Do you have a device running Android 13 (API level 33) to test the new API?